pkg/ebpf/c/tracer-ebpf.c

```diff
@@ -484,16 +430,33 @@ static int handle_message(struct sock* sk,
 
 __attribute__((always_inline))
 static int handle_retransmit(struct sock* sk, tracer_status_t* status) {
+    conn_tuple_t t = {};
     u64 ts = bpf_ktime_get_ns();
+    u64 zero = 0;
+
+    if (!read_conn_tuple(&t, status, sk, zero, CONN_TYPE_TCP)) {
+        return 0;
+    }
 
-    handle_family(sk, status, update_tcp_stats(sk, status, family, 1, ts));
+    tcp_stats_t stats = { .retransmits = 1, .rtt = 0, .rtt_var = 0 };
+    update_tcp_stats(&t, stats);
 
     // Update latest timestamp that we've seen - for connection expiration tracking
-    u64 zero = 0;
     bpf_map_update_elem(&latest_ts, &zero, &ts, BPF_ANY);
     return 0;
 }
 
+__attribute__((always_inline))
+static void handle_tcp_stats(conn_tuple_t* t, tracer_status_t* status, struct sock* sk) {
```

Comment:
In a future PR we could track the retransmits here (by also guessing the retransmit field in the TCP socket struct) and remove the associated code/kprobe.